### DISCUSSION TITLE
SFTPGo sftpd keeps dropping connection showing and showing an EOF error in server logs

### DISCUSSION THREAD

**Discussion by RealYHD** [drakkan/sftpgo#1126] [Q&A] [answered]
I've been having this issue where the SFTP daemon built in SFTPGo consistently keeps dropping connections when a file transfer occurs, but not when just browsing through the files. Due to this, I can't perform any file transfers between client and server over SFTP. File transfers over WebDAV work fine.

SSHFS-Win Manager results in:
```
stderr: client_loop: send disconnect: Connection reset by peer
```

While the server shows the following log messages:
```
stat error for path \"some/path\": lstat /srv/sftpgo/data/user/some/path: no such file or directory
...
ssh connection removed, num open ssh connections: 0
Unexpected error for transfer, path: \"/srv/sftpgo/data/user/some/path\", error: \"unexpected EOF\" bytes sent: 0, bytes received: 0 transfer running since 17 ms
...
transfer error: unexpected EOF, path: \"/srv/sftpgo/data/user/some/path.filepart\"
```

If a more detailed log is needed, I'm willing to provide. I'd also like to note that the SFTPGo server is behind a proxy. I'm also not 100% sure it's SFTPGo specific, although the normal OpenSSH server SFTP subsystem _seems_ to be fine. So really any advice would be appreciated.

**Comment by RealYHD** [comment, marked as answer]
Turns out changing the port from 21 worked, although I'm not to clear on why. Especially considering it was able to display directory content change paths just fine. I don't believe there were any other services using the port. There were also no errors on reserving port 21 on the server side of things. This may have to do with ports numbered less than 1024 being considered privileged ports? But again, nothing on the server side when opening the port. The router did change before it stopped working, but I couldn't see why this would explain the issue. If anyone has an explanation, I'd appreciate it, I'll leave the question open for a bit longer, otherwise, this question does have a solution, despite being incomplete.

**Comment by drakkan** [comment]
> error: \"unexpected EOF\" bytes sent: 0, bytes received: 0 transfer running since 17 m

You have a transfer stalled since 17 minutes. The issue seems unrelated to SFTPGo. 

Anyway, I no longer provide free support, check the [support policy](https://github.com/drakkan/sftpgo#support-policy). This is the only way to prevent SFTPGo from being discontinued/archived and it is necessary because most of people just ask and never give anything back spontaneously. Thanks for understanding

**Reply by RealYHD** [reply]
> > error: "unexpected EOF" bytes sent: 0, bytes received: 0 transfer running since 17 m
> 
> You have a transfer stalled since 17 minutes. The issue seems unrelated to SFTPGo.

That might've been me not copying the "s" character after the m, i.e, `ms`, as that's what I recall seeing, my bad.

> Anyway, I no longer provide free support, check the [support policy](https://github.com/drakkan/sftpgo#support-policy). This is the only way to prevent SFTPGo from being discontinued/archived and it is necessary because most of people just ask and never give anything back spontaneously. Thanks for understanding

No problem. Thanks.
